### PR TITLE
Improve setup and change source code layout to comply with PEP-660.

### PR DESCRIPTION
**Pull request by jlenain**
Following @maxnoe 's path on `ctapipe_io_lst`, this PR proposes to improve the `setuptools` configuration to comply with PEP-660.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,7 +1,61 @@
...
+package_dir =
+    = src
+python_requires = ~=3.8
+zip_safe = False
+install_requires =
+    astropy~=4.2
+    ctapipe~=0.12
+    numpy~=1.22
+    protozfits~=2.0
+    pytables>=3.7
```
Comment (maxnoe) [inline setup.cfg:34]: The package name on pypi is tables, without the py

Comment (maxnoe) [reply setup.cfg:34]: Conda calls it pytables though (so it is correct in environment.yml)

Comment (jlenain) [reply setup.cfg:34]: Thanks a lot, @maxnoe ! How mean...

**Comment by dkerszberg** [comment]
The change seem to work on my side. @frnbrun ca you confirm it fixed the problem you reported in https://github.com/cta-observatory/nectarchain/issues/28 ?

@jlenain I see one test failed with:
`` FAILED src/ctapipe_io_nectarcam/tests/test_version.py::test_version - AssertionError: assert '0.0.0' != `'0.0.0'`
It seems to be related with your change of version (which is a good idea I agree :-) ), can we remove the test on the version or update it to something that won't fail (I didn't check at all what is done there tbh)?

**Comment by maxnoe** [comment]
@dkerszberg the test fails because there is no tag yet in this repository. Either switch it off for the time being, or create an `0.1.0` tag on the master before merging this.

**Comment by jlenain** [comment]
Hi @dkerszberg,
Thanks for the feedback!
Indeed, as pointed out by @maxnoe , this test error will vanish as soon as we start tagging the code. I'd prefer to keep this mechanism in place, which is actually based on the implementation e.g. from `astropy`, `ctapipe`, or `ctapipe_io_lst`, since it is meant to automatize and ease releases.

**Comment by dkerszberg** [comment]
Thank you both for the clarification! I see @jlenain already clarified in the other thread that it fixes the issue from @frnbrun so I'm ready to do the merge. I'll just wait until tomorrow in case @frnbrun wants to have a look :-)

**Review by dkerszberg — APPROVED** [review]
Ready to be merged

**Comment by frnbrun** [comment]
Thanks @jlenain and @dkerszberg !

**Pull request merged**